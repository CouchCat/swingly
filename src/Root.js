import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { PersistGate } from 'redux-persist/integration/react';
import { Provider } from 'react-redux';

import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { persistStore } from 'redux-persist';
import thunk from 'redux-thunk';
// import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from './redux/root-reducer';
import { library } from '@fortawesome/fontawesome-svg-core';
import { 
  faBars,
  faSyncAlt,
  faUserCircle,
  faEdit,
  faSignOutAlt,
  faSignInAlt,
  faHome,
  faCalendar,
  faCalendarDay,
  faCalendarWeek,
  faEllipsisV,
  faCalendarPlus,
  faHatWizard,
  faMoon,
  faSun,
  faChevronLeft
} from '@fortawesome/free-solid-svg-icons';

export default (props) => {
  const { children, initialState = {} } = props;
  const middlewares = [thunk];

  if (process.env.NODE_ENV === 'development') {
    middlewares.push(logger);
  }
  
  const store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(...middlewares),
  );

  const persistor = persistStore(store);

  library.add(
    faBars,
    faSyncAlt,
    faUserCircle,
    faEdit,
    faSignOutAlt,
    faSignInAlt,
    faHome,
    faCalendar,
    faCalendarDay,
    faCalendarWeek,
    faEllipsisV,
    faCalendarPlus,
    faHatWizard,
    faMoon,
    faSun,
    faChevronLeft
  );

  return (
    <Provider store={store}>
      <BrowserRouter>
        <PersistGate persistor={persistor}>
          {children}
        </PersistGate>
      </BrowserRouter>
    </Provider>
  );
}