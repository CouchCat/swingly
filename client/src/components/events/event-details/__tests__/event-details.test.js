import React from 'react';
import ConnectedEventDetails, { EventDetails } from '../event-details.component';
import { shallow, mount } from 'enzyme';
import Root from '../../../../Root';
import renderer from 'react-test-renderer';

xdescribe('Event Details Component', () => {
  let wrapper = null;
  let mockProps = null;
  let mockOnClose = null;
  let mockEvent = null;
  
  beforeEach(() => {
    mockOnClose = jest.fn();
    mockEvent = {
      cancelled: false,
      courses: [],
      currency: "euro",
      description: "We're back with a newly-construct­ed floor. Drop on in for a dance in the green and glorious surroundings of Jockel Biergarten, very near the Landwehrkanal in Kreuzberg. There's beer, lemonade, and food there, not to mention plenty of sun- and all the fresh outdoor air you can breathe. (There is a bit of shade available as well.)↵↵There are both indoor and outdoor spaces to dance, so we can dance in all weather conditions.↵↵With DJ! Toss a bit of cash in the hat: your generous donations help keep the whole flying outdoor circus running.",
      end: "2019-09-01T18:00:00.000Z",
      going: [],
      host: ["A8WnU1fmQnWQ3bjvI6PYro32Tjh1"],
      id: "6rfrXIvwXLsEWdzvvSvw",
      interested: [],
      links: [],
      location: "Jockel Biergarten, Ratiborstraße 14C, 10999 Berlin, Germany (Karte)",
      mapLink: "https://maps.google.com/maps?hl=de&q=Jockel%20Biergarten%2C%20Ratiborstra%C3%9Fe%2014C%2C%2010999%20Berlin%2C%20Germany",
      otherFees: [],
      start: "2019-09-01T14:00:00.000Z",
      title: "Outdoors!",
      type: "social",
    };

    mockProps = {
      isOpen: true,
      onClose: mockOnClose,
      event: mockEvent
    };

    const initialState = {
      themeMode: { darkMode: false }
    }
    
    wrapper = mount(
      <Root initialState={initialState}>
        <ConnectedEventDetails {...mockProps} />
      </Root>
    );
  });

  it('should render correctly', () => {
    mockProps = {
      isOpen: true,
      onClose: mockOnClose,
      event: mockEvent
    };

    const tree = renderer.create(<EventDetails {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});