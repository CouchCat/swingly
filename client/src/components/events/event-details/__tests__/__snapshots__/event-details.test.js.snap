// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Details Component should render correctly 1`] = `
Array [
  <button
    className="sc-kpOJdX gGjbXx"
    onClick={[MockFunction]}
  />,
  <div
    className="sc-dxgOiQ ePLTYJ"
  >
    <div
      className="css-1smlpex"
    >
      <div
        className="css-18xo8fn"
      />
      <div
        className="css-gfx32u"
      />
      <div
        className="css-clle6u"
      />
      <div
        className="css-146uy45"
      />
      <div
        className="css-raj17w"
      />
      <div
        className="css-raicj8"
      />
    </div>
  </div>,
]
`;
